src/main/java/com/flippingutilities/model/AccountData.java

```diff
@@ -98,6 +100,21 @@ public void prepareForUse(FlippingPlugin plugin)
 		}
 	}
 
+	/**
+	 * When a user is an f2p world and recieves events for members items (cause they were already in the GE),
+	 * the item manager retrieves the item's name as "Members object". The item manager returns the correct
+	 * name when the user is on a member's world or logged out. As such, this method is called when the plugin starts
+	 * and whenever the user logs into a members world to clean up any "Members object" item names.
+	 */
+	public void fixIncorrectItemNames(ItemManager itemManager) {
+		trades.forEach(item -> {
+			if (item.getItemName().equals("Members object")) {
+				String actualName = itemManager.getItemComposition(item.getItemId()).getName();
```

Comment:
this will trigger only on plugin startup and when you log into a member world. 

If you are on an f2p world, the itemManager returns "Members object" for all members items, so there is no way to set it when they are on an f2p world.